bitmap_amd64.go

```diff
@@ -1,69 +1,74 @@
-// Copyright (c) Roman Atachiants and contributors. All rights reserved.
-// Licensed under the MIT license. See LICENSE file in the project root for details.
-
-package bitmap
-
-import (
-	"github.com/klauspost/cpuid/v2"
-)
-
-var (
-	avx2 = cpuid.CPU.Supports(cpuid.AVX2)
-	popc = cpuid.CPU.Supports(cpuid.POPCNT)
-)
-
-// And computes the intersection between two bitmaps and stores the result in the current bitmap
-func (dst *Bitmap) And(b Bitmap) {
-	if dst.balance(b); len(*dst) >= len(b) {
-		switch avx2 {
-		case true:
-			x64and(*dst, b)
-		default:
-			and(dst, b)
-		}
-	}
-}
-
-// AndNot computes the difference between two bitmaps and stores the result in the current bitmap
-func (dst *Bitmap) AndNot(b Bitmap) {
-	if dst.balance(b); len(*dst) >= len(b) {
-		switch avx2 {
-		case true:
-			x64andn(*dst, b)
-		default:
-			andn(dst, b)
-		}
-	}
-}
-
-// Or computes the union between two bitmaps and stores the result in the current bitmap
-func (dst *Bitmap) Or(b Bitmap) {
-	if dst.balance(b); len(*dst) >= len(b) {
-		switch avx2 {
-		case true:
-			x64or(*dst, b)
-		default:
-			or(dst, b)
-		}
-	}
-}
-
-// Xor computes the symmetric difference between two bitmaps and stores the result in the current bitmap
-func (dst *Bitmap) Xor(b Bitmap) {
-	if dst.balance(b); len(*dst) >= len(b) {
-		switch avx2 {
-		case true:
-			x64xor(*dst, b)
-		default:
-			xor(dst, b)
-		}
-	}
-}
-
-// Count returns the number of elements in this bitmap
-func (dst Bitmap) Count() int {
-	if popc {
-		return int(x64count(dst))
-	}
-	return count(dst)
-}
+// Copyright (c) Roman Atachiants and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+package bitmap
+
+import (
+	"github.com/klauspost/cpuid/v2"
+)
+
+var (
+	avx2 = cpuid.CPU.Supports(cpuid.AVX2)
+	popc = cpuid.CPU.Supports(cpuid.POPCNT)
+)
+
+// And computes the intersection between two bitmaps and stores the result in the current bitmap
+func (dst *Bitmap) And(b Bitmap) {
+	var min int
+	if len(b) >= len(*dst) {
```

Comment:
This changes the behavior, which is fine but we probably ought to be consistent and also do the same thing in `AndNot`, Or` and `Xor`

Reply:
I have updated PR with fixes for `AndNot`, `Or` and `Xor`, but might be in a different way that you expect. Could you clarify more?

Reply:
It's correct now I think, `Or` and `Xor` only require growing, since `Or` is essentially a union.